### ISSUE TITLE
False positive `unspecified-encoding` for Path objects opened in binary mode

### ISSUE THREAD

**Issue by jacobtylerwalls** [PyCQA/pylint#5017]
### Bug description

```python
'''
This snippet raises no warnings in previous versions, but raises unspecified-encoding in 2.11.0
Opening in binary mode -- either 'wb' or 'rb'.
'''
import pathlib
fp = pathlib.Path(__file__)
with fp.open('rb') as f:
    pass
```

### Command used

```shell
pylint a.py
```

### Pylint output

```shell
************* Module a
a.py:6:5: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
```

### Expected behavior

none

### Pylint version

```shell
pylint 2.11.0
astroid 2.8.0
Python 3.9.6 (v3.9.6:db3ff76da1, Jun 28 2021, 11:49:53) 
[Clang 6.0 (clang-600.0.57)]
```

### OS / Environment

macOS

**Comment by Pierre-Sassoulas** [comment]
@DanielNoord do you want to handle this ?
